src/conn/conn_layered.c

```diff
@@ -1129,6 +1129,73 @@ __disagg_restart_checkpoint(WT_SESSION_IMPL *session)
     return (ret);
 }
 
+/*
+ * __disagg_step_up --
+ *     Step up to the node to the leader mode.
+ */
+static int
+__disagg_step_up(WT_SESSION_IMPL *session)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+
+    conn = S2C(session);
+
+    /*
+     * We need to hold the checkpoint lock while stepping up, because if we change the role
+     * concurrently with a checkpoint, it would do only a part of the work required for the new
+     * role, leaving the database in an inconsistent state.
+     */
+    WT_ASSERT_SPINLOCK_OWNED(session, &conn->checkpoint_lock);
+
+    F_SET(conn, WT_CONN_RECONFIGURING_STEP_UP);
+
+    /* Step up to the leader mode. */
+    conn->layered_table_manager.leader = true;
+    WT_STAT_CONN_SET(session, disagg_role_leader, 1);
```

Comment:
Would it be better to move these two steps after draining the ingest stable? That way we know for certain these were set after the necessary steps succeeded.